Framework/Kernel/src/TimeROI.cpp

```diff
@@ -185,13 +187,15 @@ void TimeROI::addMask(const Types::Core::DateAndTime &startTime, const Types::Co
   } else {
     g_log.debug("TimeROI::addMask cutting notch in existing ROI");
     // create rois for before and after the mask
-    TimeInterval use_before(m_roi.front(), startTime);
-    TimeInterval use_after(stopTime, m_roi.back());
+    const TimeInterval use_before(m_roi.front(), startTime);
+    const TimeInterval use_after(stopTime, m_roi.back());
 
     // loop through all current splitters and get new intersections
     std::vector<TimeInterval> output;
     TimeInterval intersection;
-    for (const auto &interval : this->toTimeIntervals()) {
```

Comment:
That wouldn't help this particular case since `TimeRoI::addMask()` is being called multiple times by the intersection function. The cache would be invalidated before it could be used.